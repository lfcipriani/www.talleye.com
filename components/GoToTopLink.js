import i from '../content/i18n';

export default function GoToTopLink({ size }) {
  return (
    <>
      <a href="#top" title={i('scrollToTop')}>
        <span className="visually-hidden">{i('scrollToTop')}</span>
        <svg
          xmlns="http://www.w3.org/2000/svg"
          width={size || '35'}
          height={size || '35'}
          fill="#0e1007"
          viewBox="0 0 256 256"
          className="svgIcon"
          aria-hidden="true"
          focusable="false"
        >
          <rect width="256" height="256" fill="none"></rect>
          <circle
            cx="128"
            cy="128"
            r="96"
            fill="none"
            stroke="#0e1007"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="16"
            className="stroke"
          ></circle>
          <polyline
            points="96 168 128 136 160 168"
            fill="none"
            stroke="#0e1007"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="16"
            className="stroke"
          ></polyline>
          <polyline
            points="96 112 128 80 160 112"
            fill="none"
            stroke="#0e1007"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="16"
            className="stroke"
          ></polyline>
        </svg>
      </a>
      <style jsx>{`
        .stroke {
          stroke: var(--color-icon);
        }
        .fill {
          fill: var(--color-icon);
        }
      `}</style>
    </>
  );
}
