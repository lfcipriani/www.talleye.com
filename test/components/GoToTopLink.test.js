import renderer from 'react-test-renderer';
import GoToTopLink from '../../components/GoToTopLink';

describe('GoToTopLink snapshot', () => {
  it('renders without size', () => {
    const component = renderer.create(<GoToTopLink />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with size', () => {
    const component = renderer.create(<GoToTopLink size="42" />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
