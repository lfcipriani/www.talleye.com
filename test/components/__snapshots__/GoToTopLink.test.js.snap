// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoToTopLink snapshot renders with size 1`] = `
<a
  className="jsx-3248159136"
  href="#top"
  title="Scroll to the top of the page"
>
  <span
    className="jsx-3248159136 visually-hidden"
  >
    Scroll to the top of the page
  </span>
  <svg
    aria-hidden="true"
    className="jsx-3248159136 svgIcon"
    fill="#0e1007"
    focusable="false"
    height="42"
    viewBox="0 0 256 256"
    width="42"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      className="jsx-3248159136"
      fill="none"
      height="256"
      width="256"
    />
    <circle
      className="jsx-3248159136 stroke"
      cx="128"
      cy="128"
      fill="none"
      r="96"
      stroke="#0e1007"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="16"
    />
    <polyline
      className="jsx-3248159136 stroke"
      fill="none"
      points="96 168 128 136 160 168"
      stroke="#0e1007"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="16"
    />
    <polyline
      className="jsx-3248159136 stroke"
      fill="none"
      points="96 112 128 80 160 112"
      stroke="#0e1007"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="16"
    />
  </svg>
</a>
`;

exports[`GoToTopLink snapshot renders without size 1`] = `
<a
  className="jsx-3248159136"
  href="#top"
  title="Scroll to the top of the page"
>
  <span
    className="jsx-3248159136 visually-hidden"
  >
    Scroll to the top of the page
  </span>
  <svg
    aria-hidden="true"
    className="jsx-3248159136 svgIcon"
    fill="#0e1007"
    focusable="false"
    height="35"
    viewBox="0 0 256 256"
    width="35"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      className="jsx-3248159136"
      fill="none"
      height="256"
      width="256"
    />
    <circle
      className="jsx-3248159136 stroke"
      cx="128"
      cy="128"
      fill="none"
      r="96"
      stroke="#0e1007"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="16"
    />
    <polyline
      className="jsx-3248159136 stroke"
      fill="none"
      points="96 168 128 136 160 168"
      stroke="#0e1007"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="16"
    />
    <polyline
      className="jsx-3248159136 stroke"
      fill="none"
      points="96 112 128 80 160 112"
      stroke="#0e1007"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="16"
    />
  </svg>
</a>
`;
